### PR TITLE
tox --version now shows version info of registered plugins

### LINKED ISSUE #544: tox --version should display loaded plugins
It seems that currently tox is very secretive about which plugins it did load and that's not giving the best user experience.

Based on the implementations from other tools, usually they display plugin information alongside with their versions, because that results describes the best what you really run.

One nice example is ``pytest --version``:
```
pytest --version                                                                                                                                                              [15:24:20]
This is pytest version 3.0.7, imported from /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/pytest.pyc
setuptools registered plugins:
  pytest-xdist-1.15.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/xdist/boxed.py
  pytest-xdist-1.15.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/xdist/looponfail.py
  pytest-xdist-1.15.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/xdist/plugin.py
  pytest-pep8-1.0.6 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/pytest_pep8.py
  pytest-mock-1.6.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/pytest_mock.py
  pytest-instafail-0.3.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/pytest_instafail.py
  pytest-cov-2.4.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/pytest_cov/plugin.py
  flaky-3.3.0 at /Users/ssbarnea/.venv/rh/lib/python2.7/site-packages/flaky/flaky_pytest_plugin.py
```

### PR DESCRIPTION
**Pull request by obestwalter**
Closes #544 

output from my box, with installed plugins:

    tox --version
    2.8.2.dev27 imported from /home/oliver/work/tox/tox/tox/__init__.py
    registered plugins:
        tox-virtualenv-no-download-1.0.0 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_virtualenv_no_download.py
        tox-travis-0.8 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_travis/hooks.py
        tox-run-command-0.4 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_run_command.py
        tox-pyenv-1.1.0 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_pyenv.py
        tox-py-backwards-0.1 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_py_backwards.py
        tox-pip-extensions-1.1.0 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_pip_extensions.py
        tox-globinterpreter-0.3 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_globinterpreter.py
        tox-docker-1.0.0 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_docker.py
        tox-bitbucket-status-1.0 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/tox_bitbucket_status.py
        tox-battery-0.5 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/toxbat/requirements.py

### CODE REVIEW

**Comment by obestwalter** [comment]
source = getattr(mod, '__file__', repr(mod))

is in case the plugin registered is not a normal module with a file attr - which is possible, I guess. I checked out how pytest does this and that safeguard is there also, so I added it here.

**Comment by obestwalter** [comment]
forgot to add towncrier thigy ... will do that later. Have to get used to that first.

File: newsfragments/544.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+``tox --version`` now shows information about all registered plugins - by :user:`obestwalter`
```
Comment (nicoddemus) [inline newsfragments/544.feature]: Interesting, does this `by :user:`obestwalter` trick work?

Comment (gaborbernat) [reply newsfragments/544.feature]: yes it does 👍

File: tox/config.py [hunk 2]
```diff
@@ -258,11 +260,16 @@ def feedback(msg, sysexit=False):
...
-        print("%s imported from %s" % (version, tox.__file__))
-        raise SystemExit(0)
+def print_version_info(pm):
+    print("%s imported from %s" % (tox.__version__, tox.__file__))
+    plugin_dist_info = pm.list_plugin_distinfo()
+    if not plugin_dist_info:
+        return
+    print('registered plugins:')
+    for mod, egg_info in plugin_dist_info:
+        source = getattr(mod, '__file__', repr(mod))
```
Comment (RonnyPfannschmidt) [inline tox/config.py]: please take a look at trimming this down

site packages installed ones shouldn't list at all
develop installed ones should point to the project folder

Comment (obestwalter) [reply tox/config.py]: I don't understand what you mean. Also: AFAK it shows pretty much the same like `pytest --version`.

Comment (RonnyPfannschmidt) [reply tox/config.py]: then we should fix it for pytest, too

Comment (obestwalter) [reply tox/config.py]: I still don't understand what you mean, though, so I can't fix anything yet :) - if I do I will fix it here and open a PR on pytest also.

Comment (RonnyPfannschmidt) [reply tox/config.py]: we shouldnt print the site packages location for each plugin, instead its enough to know its in site pacakges usually.

only for the eggs it important to know where they where added

i do wonder if we should perhaps put the utility for listing plug-ins and their versions/locations into pluggy

Comment (obestwalter) [reply tox/config.py]: ah - now I get it - the paths. I actually do not have a problem with showing the real path to where it is imported from, just like for tox itself.

Comment (obestwalter) [reply tox/config.py]: I like the idea of having this the same over all tools that use pluggy, so I will open an issue there and gladly use that function of pluggy, when it is introduced.
 
For here and now, I merge this as it in as this is exactly parallel to what pytest outputs with the difference that pytest outputs to stderr (I don't know why) and tox outputs to stdout.
`pytest --version`:

    This is pytest version 3.2.2, imported from /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/pytest.py
    setuptools registered plugins:
      pytest-cov-2.5.1 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/pytest_cov/plugin.py

**Comment by obestwalter** [comment]
made it easier to test now and added some actual tests.

**Comment by asottile** [comment]
@obestwalter if you run `.tox/style/bin/pre-commit install`, it'll run automatically on each commit :)

**Comment by obestwalter** [comment]
Thanks @asottile - did that. That's a really great tool.

File: newsfragments/544.feature (hunk 1 above)
Comment (nicoddemus) [reply newsfragments/544.feature]: Interesting, didn't find this in the docs, I will take a look at the code when I have the time to figure out what other tricks are hidden there.

**Pull request merged**